### ISSUE TITLE
association of registry entries with protocol or organization?

### ISSUE THREAD

**Issue by dret** [dret/I-D#28]
comment from john curran:

    2) You don’t highlight the need to be exceptionally clear about what is being 
        associated with the registry values…  for example, a specific UDP port may
        be reserved by a company for protocol “foo”, but it is the protocol that gets
        the reservation, not the company.    There are registries (e.g. the general 
        portions of the AS and IP address spaces, private enterprise number, etc.)
        where the value has no protocol meaning aside from uniqueness, and is
        truly associated with an organization.   Protocol designers who specify such
        new general purpose registries should do so with significant care and clarity,
        as the administration of these registries is often more complex that typical.

**Comment by dret** [comment]
@john5223, i think i understand your concern. but this really is about the semantics of registry contents, right? in this case it's a special kind of semantics because it is about the fact whether the value is indeed self-describing (i.e., when i see it i know what it means), or just reserved (when i see it, i have to follow some process to find out what it means). is that the distinction you are referring to?

**Comment by jcurranz** [comment]
No, I'm thinking of something much more basic... can someone reserve an entry in IANA registry and sell it to another party?  If the assignment of an entry is to the specification of that particular option value, then the assignment is to the purpose.  If the assignment of an entry is to the requesting party, then the answer is far less clear.

The IETF has had occasions where other parties (e.g. another standard body) just said "just give us a entry in that registry, and we'll define what it means at some future time when we finish the standard"

**Comment by dret** [comment]
> The IETF has had occasions where other parties (e.g. another standard
> body) just said "just give us a entry in that registry, and we'll define
> what it means at some future time when we finish the standard"

now that you're mentioning it, 
https://twitter.com/dret/status/699335652572295173 was something i 
thought about just today: isn't DNS a registry of some kind? and if 
that's the case, then what you're saying here makes a lot of sense in 
terms of people buying domain names, and then selling them at a profit.

**Comment by jcurranz** [comment]
Right... The point is that some registries are simply uniqueness values, which implies that the registrant may potentially transfer their assignment, whereas others are assigned to particular code points as represented by extensions to the standard.

**Comment by dret** [comment]
On 2016-02-15 23:31 , John Curran wrote:
> No, I'm thinking of something much more basic... can someone reserve an
> entry in IANA registry and sell it to another party? If the assignment
> of an entry is to the specification of that particular option value,
> then the assignment is to the purpose. If the assignment of an entry is
> to the requesting party, then the answer is far less clear.

i completely agree. but in the end, that's a question of the entry 
semantics, right?

* if the registry value should identify "protocol meaning", then the 
entry itself must be associated with an entry establishing that meaning.

* if the registry value only is a unique value, and the protocol meaning 
has to be discovered independently of the registry, then all the 
registry can do is point to the owner of the entry, and discovering the 
value's meaning is out-of-band from the registry point of view.

> The IETF has had occasions where other parties (e.g. another standard
> body) just said "just give us a entry in that registry, and we'll define
> what it means at some future time when we finish the standard"

i'd say is up for the registry to decide whether that's acceptable or 
not. but if that's allowed then it certainly violates some principles of 
RFC 7500, because then there's no stability in the value's meaning, 
which probably makes such a registry not quite as useful as it could be.

**Comment by dret** [comment]
On 2016-02-15 23:45 , John Curran wrote:
> Right... The point is that some registries are simply uniqueness values,
> which implies that the registrant may potentially transfer their
> assignment, whereas others are assigned to particular code points as
> represented by extensions to the standard.

i think we're in agreement here. but if a registry only guarantees 
uniqueness, then that's a very weak registry, because some of the 
principles of RFC 7500 are not adhered to.

but in some cases that may be enough. if you look at DNS domain names, 
then maybe it's good enough that all the registry (the DNS) guarantees 
is that a value can be resolved. that's all that matters for this 
particular registry, that i can ask at some point in time which host 
address is associated with that entry value. this may change over time, 
but in this case that's actually the goal, and DNS makes sure that i 
have an infrastructure that can handle this scenario well. but in this 
scenario it's a feature than assignments can change over time.

**Comment by jcurranz** [comment]
My principal concern is that there is clarity in any draft that proposes a registry regarding the entry semantics.
